### PR TITLE
chore: Tile a11y with anchors

### PR DESCRIPTION
**Pull request by gretanausedaite**
Solving `Tile` part of this issue: #840 

#### Added:
- `iui-link-box` and `iui-link-action` class to add pseudocontent to action component.
- Added LinkBox component for wrapper class
- Added LinkAction component for pseudocontent
- Tile.Action component which should be used in actionable Tiles.

#### Testing:
Unit, visual tests.
Manual testing with screen reader

#### Docs:
Jsdocs,

### CODE REVIEW

File: packages/itwinui-react/src/core/Tile/TileAction.tsx [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+ * Polymorphic action component for actionable Tiles. Should be used in name.
+ * It is `button` by default. Can also be rendered as anchor link.
+ *
+ * @example
+ * <Tile
+ *   isActionable={isActionable}
+ *   name={<Tile.Action onClick={() => doOnClick()}>Tile name</Tile.Action>}
+ * />
+ */
+export const TileAction = <T extends React.ElementType = 'button'>(
```
Comment (mayank99) [inline packages/itwinui-react/src/core/Tile/TileAction.tsx:35]: we have a utility for polymorphic component. you can see usage in `Flex.tsx`

File: packages/itwinui-react/src/core/Tile/TileAction.tsx [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  props: TileActionProps<T>,
+) => {
+  const { as: Element = 'button', className, children, ...rest } = props;
+  return (
+    <Element
+      tabIndex={-1}
```
[the remark this answers is not in the document]
Comment (mayank99) [reply packages/itwinui-react/src/core/Tile/TileAction.tsx:41]: i'm also wondering if this should be a generic utility component, rather than specific to Tile

File: packages/itwinui-css/src/tile/tile.scss [hunk 3]
```diff
@@ -73,6 +73,25 @@
         transform: translateZ(0) scale(1.1);
       }
     }
+
+    position: relative;
+
+    .iui-tile-content {
+      position: unset;
```
[the remark this answers is not in the document]
Comment (mayank99) [reply packages/itwinui-css/src/tile/tile.scss:80]: I think we need to support buttons. It's the reason I abandoned this PR: https://github.com/iTwin/iTwinUI/pull/994/files

File: packages/itwinui-react/src/core/Tile/TileAction.tsx (hunk 1 above)
Comment (gretanausedaite) [reply packages/itwinui-react/src/core/Tile/TileAction.tsx]: Using polymorphic util component.

File: packages/itwinui-react/src/core/Tile/TileAction.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (gretanausedaite) [reply packages/itwinui-react/src/core/Tile/TileAction.tsx]: Moved to generic component. I thought about it too as it might be reused in MenuItem.

File: packages/itwinui-css/src/tile/tile.scss [hunk 4]
```diff
@@ -73,6 +73,12 @@
         transform: translateZ(0) scale(1.1);
       }
     }
+
+    position: relative;
+
+    // .iui-tile-content {
+    //   position: unset;
+    // }
```
Comment (FlyersPh9) [inline packages/itwinui-css/src/tile/tile.scss:81]: Don't forget to remove this before merging.

File: packages/itwinui-css/src/utils/component-action/component-action.scss [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+/// If it contains a focusable element, make sure to revert/exclude these styles.
+@mixin component-action {
+  all: unset;
+  text-decoration: none;
+  color: inherit;
+  user-select: text;
+
+  &::after {
+    content: '';
+    position: absolute;
```
Comment (FlyersPh9) [inline packages/itwinui-css/src/utils/component-action/component-action.scss:15]: This `@mixin component-action` is an interesting idea, I like it!  Does it make sense to put `position: relative` above the pseudo `::after`?

File: packages/itwinui-react/src/core/utils/components/PolymorphicAction.tsx [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+  'a',
+  PolymorphicActionOwnProps
+>;
+
+/**
+ * Polymorphic action component.
+ * It is rendered as `a` by default.
+ *
+ */
+export const PolymorphicAction = React.forwardRef((props, ref) => {
```
Comment (mayank99) [inline packages/itwinui-react/src/core/utils/components/PolymorphicAction.tsx:29]: Need to think about the naming. `PolymorphicAction` is too low-level, and even the JSDoc explains nothing.

Maybe you could list out any other names that you can think of, and do some research into other libraries? One that I've found is `LinkBox` + `LinkOverlay` from chakra-ui. 
https://chakra-ui.com/docs/components/link-overlay

File: packages/itwinui-react/src/core/utils/components/PolymorphicAction.tsx [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+import {
+  PolymorphicComponentProps,
+  PolymorphicForwardRefComponent,
+} from '../index';
+
+type PolymorphicActionOwnProps = {
+  /**
+   * Content in the action element.
+   */
+  children?: React.ReactNode;
```
Comment (mayank99) [inline packages/itwinui-react/src/core/utils/components/PolymorphicAction.tsx:16]: `children` doesn't need to be a special prop.

File: packages/itwinui-react/src/core/utils/components/PolymorphicAction.tsx [hunk 8]
```diff
@@ -0,0 +1,44 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import React from 'react';
+import cx from 'classnames';
+import {
+  PolymorphicComponentProps,
+  PolymorphicForwardRefComponent,
+} from '../index';
```
Comment (mayank99) [inline packages/itwinui-react/src/core/utils/components/PolymorphicAction.tsx:10]: we should never import anything from `index` to avoid risking circular dependencies.

```suggestion
} from '../props';
```

File: packages/itwinui-react/src/core/Tile/Tile.tsx [hunk 9]
```diff
@@ -137,148 +138,165 @@ export type TileProps = {
...
+     * Polymorphic action component for actionable Tiles. Should be used in name.
+     * It is `button` by default. Can also be rendered as anchor link.
+     *
+     * @example
+     * <Tile
+     *   isActionable={isActionable}
+     *   name={<Tile.Action onClick={() => doOnClick()}>Tile name</Tile.Action>}
+     * />
+     */
+    Action: PolymorphicAction,
```
Comment (mayank99) [inline packages/itwinui-react/src/core/Tile/Tile.tsx:295]: does it need to be a subcomponent now? i think users could just pass the generic component directly

File: packages/itwinui-css/src/utils/component-action/component-action.scss [hunk 10]
```diff
@@ -0,0 +1,18 @@
+// Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+// See LICENSE.md in the project root for license terms and full copyright notice.
+
+/// Visually hides an element but keeps it accessible to screen readers.
+/// Because it applies some potentially invasive styles, use this mixin on a wrapper element for best results.
+/// If it contains a focusable element, make sure to revert/exclude these styles.
```
Comment (mayank99) [inline packages/itwinui-css/src/utils/component-action/component-action.scss:6]: these comments have nothing to do with the mixin :/

File: packages/itwinui-css/src/utils/component-action/component-action.scss [hunk 11]
```diff
@@ -0,0 +1,18 @@
...
+@mixin component-action {
+  all: unset;
+  text-decoration: none;
+  color: inherit;
+  user-select: text;
```
Comment (mayank99) [inline packages/itwinui-css/src/utils/component-action/component-action.scss:11]: does `user-select` do anything here?

File: packages/itwinui-react/src/core/Tile/TileAction.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (mayank99) [reply packages/itwinui-react/src/core/Tile/TileAction.tsx]: tabindex still needs to be fixed

File: packages/itwinui-css/src/utils/component-action/component-action.scss (hunk 5 above)
Comment (mayank99) [reply packages/itwinui-css/src/utils/component-action/component-action.scss:15]: `position: relative` will go on the wrapper (Tile or MenuItem), not on the link.

File: packages/itwinui-react/src/core/Tile/TileAction.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (mayank99) [reply packages/itwinui-react/src/core/Tile/TileAction.tsx]: after that, i expect we will need to update focus styling to use `:has` or `:focus-within`

File: apps/storybook/src/Tile.stories.tsx [hunk 12]
```diff
@@ -197,6 +197,41 @@ Actionable.args = {
...
+  );
+};
+AnchorLink.argTypes = { ...Basic.argTypes };
+AnchorLink.args = {
+  ...Basic.args,
+  description:
+    'If you click on this stadium, it is going to open another page.',
+  isActionable: true,
+  moreOptions: undefined,
+};
```
Comment (elephantcatdog) [inline apps/storybook/src/Tile.stories.tsx:234]: Could you make this example actually "open" a new page, even if it's just the same page? In this storybook example, the text says "If you click on this stadium, it is going to open another page.", but it seems broken because nothing happens when it's clicked on. It could be like the links in the story fo `SkipToContentLink` or like in `Anchor - Basic`.

**Review by elephantcatdog — COMMENTED** [review]
Not super related, but are the css visual tests 'with thumbnail' meaningful if we're not testing them with thumbnails? We're hiding the thumbnails because it caused test problems previously.

File: packages/itwinui-css/src/utils/component-action/component-action.scss (hunk 11 above)
Comment (gretanausedaite) [reply packages/itwinui-css/src/utils/component-action/component-action.scss:11]: Yes, without it text gets selected on click.

File: packages/itwinui-css/src/utils/component-action/component-action.scss (hunk 10 above)
Comment (gretanausedaite) [reply packages/itwinui-css/src/utils/component-action/component-action.scss]: removed

File: packages/itwinui-react/src/core/utils/components/PolymorphicAction.tsx (hunk 8 above)
Comment (gretanausedaite) [reply packages/itwinui-react/src/core/utils/components/PolymorphicAction.tsx]: importing from props.

File: packages/itwinui-react/src/core/utils/components/PolymorphicAction.tsx (hunk 7 above)
Comment (gretanausedaite) [reply packages/itwinui-react/src/core/utils/components/PolymorphicAction.tsx]: It does not work without children so I kept it in props.

File: packages/itwinui-react/src/core/utils/components/PolymorphicAction.tsx (hunk 6 above)
Comment (gretanausedaite) [reply packages/itwinui-react/src/core/utils/components/PolymorphicAction.tsx:21]: `ActionOverlay`? I don't want to blindly copy from other libraries. Maybe we need team vote?

File: packages/itwinui-css/src/utils/component-action/component-action.scss (hunk 11 above)
Comment (mayank99) [reply packages/itwinui-css/src/utils/component-action/component-action.scss:8]: This might be because `iui-tile-name` has `user-select: all`. We should remove that.

File: packages/itwinui-react/src/core/utils/components/PolymorphicAction.tsx (hunk 7 above)
Comment (mayank99) [reply packages/itwinui-react/src/core/utils/components/PolymorphicAction.tsx]: what doesn't work without children? if you check `Flex.tsx`, none of them have explicitly defined children.

i think you can just use `{}`

File: packages/itwinui-css/src/utils/component-action/component-action.scss (hunk 11 above)
Comment (gretanausedaite) [reply packages/itwinui-css/src/utils/component-action/component-action.scss]: Removed select all

File: packages/itwinui-react/src/core/utils/components/PolymorphicAction.tsx (hunk 6 above)
Comment (gretanausedaite) [reply packages/itwinui-react/src/core/utils/components/PolymorphicAction.tsx]: Added two components. LinkBox has tabIndex to skip it when tabbing.

Comment (mayank99) [reply packages/itwinui-react/src/core/utils/components/PolymorphicAction.tsx]: The naming is kinda reversed. Box should the outer wrapper, and Overlay should be inner one.

And why do you want to skip it when tabbing?

File: packages/itwinui-css/src/tile/tile.scss (hunk 3 above)
[the remark this answers is not in the document]
Comment (mayank99) [reply packages/itwinui-css/src/tile/tile.scss]: #1117 has been merged so this should be possible to achieve now

[the remark this answers is not in the document]
Comment (gretanausedaite) [reply packages/itwinui-css/src/tile/tile.scss]: Removed tile-content css part.

File: packages/itwinui-css/src/tile/tile.scss (hunk 4 above)
Comment (gretanausedaite) [reply packages/itwinui-css/src/tile/tile.scss]: Removed :)

File: packages/itwinui-react/src/core/utils/components/PolymorphicAction.tsx (hunk 6 above)
Comment (gretanausedaite) [reply packages/itwinui-react/src/core/utils/components/PolymorphicAction.tsx]: Reversed the naming.

In the article linked in the issue, when you tab through the tiles it tabs only once per tile. I think it's more smooth behavior - click on full tile to open a link, not same action in two places. I could be wrong.

File: apps/storybook/src/Tile.stories.tsx (hunk 12 above)
Comment (gretanausedaite) [reply apps/storybook/src/Tile.stories.tsx:241]: Added link to story in docs tab.

File: apps/storybook/src/Tile.stories.tsx [hunk 13]
```diff
@@ -197,6 +197,41 @@ Actionable.args = {
...
+    thumbnail,
+    isNew,
+    isActionable = true,
+    ...rest
+  } = props;
+
+  return (
+    <Tile
+      name={
+        <Tile.Action onClick={() => console.log('clicked')}>{name}</Tile.Action>
```
[the remark this answers is not in the document]
Comment (gretanausedaite) [reply apps/storybook/src/Tile.stories.tsx]: Reworked this part of code.

File: packages/itwinui-react/src/core/utils/components/PolymorphicAction.tsx (hunk 6 above)
Comment (gretanausedaite) [reply packages/itwinui-react/src/core/utils/components/PolymorphicAction.tsx]: Removed tabIndex from Tile, should be good now.

File: packages/itwinui-react/src/core/Tile/TileAction.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (gretanausedaite) [reply packages/itwinui-react/src/core/Tile/TileAction.tsx]: Updated styles for tile focus when selected state.

File: packages/itwinui-react/src/core/utils/components/PolymorphicAction.tsx (hunk 7 above)
Comment (gretanausedaite) [reply packages/itwinui-react/src/core/utils/components/PolymorphicAction.tsx]: I can use `{}` with eslint disabled

Comment (mayank99) [reply packages/itwinui-react/src/core/utils/components/PolymorphicAction.tsx]: `{}` can be assigned to a different type for better readability


this also makes it easier to add more props later if needed

File: packages/itwinui-css/src/utils/component-action/link-box.scss [hunk 14]
```diff
@@ -0,0 +1,26 @@
+// Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+// See LICENSE.md in the project root for license terms and full copyright notice.
+@import '../mixins';
+
+@mixin iui-link-box {
+  position: relative;
+  width: fit-content;
+  height: fit-content;
+  cursor: pointer;
```
Comment (mayank99) [inline packages/itwinui-css/src/utils/component-action/link-box.scss:9]: `cursor: pointer` should go on link-overlay.

`width:fit-content` and `height: fit-content` shouldn't be necessary. This class should have no opinion on the sizing. but i think you added this because `div`s automatically take up the available width. This can be fixed using the `as` prop.
```
<LinkBox as={Tile}>
```
```
<div class='iui-tile iui-link-box'>
```

however, in #1110, I've also added `position: relative` to the Tile, so maybe there is no need to even wrap the Tile in Linkbox.

File: packages/itwinui-css/src/utils/component-action/link-box.scss [hunk 15]
```diff
@@ -0,0 +1,26 @@
...
+
+  &:where(:focus-within) {
```
Comment (mayank99) [inline packages/itwinui-css/src/utils/component-action/link-box.scss:11]: the focus styling looks good for the most simple case when tabbing, but it will also show for mouse users. the solution there is to use `:has(:focus-visible)` and then add `:focus-within` fallback for browsers that don't support `:has`.

for more advanced cases, it looks a bit weird. am i focusing the tile (main action) or the secondary action inside it?


this can be fixed using `&:has(.iui-link-overlay:focus-visible)` but the fallback approach for this will still have the same problem. maybe that's ok? firefox will support `:has` by Q2 2023.

File: packages/itwinui-css/src/utils/component-action/link-box.scss [hunk 16]
```diff
@@ -0,0 +1,26 @@
...
+    outline: var(--iui-size-3xs) solid var(--iui-color-border-accent);
+    border-radius: var(--iui-border-radius-1);
```
Comment (mayank99) [inline packages/itwinui-css/src/utils/component-action/link-box.scss:13]: border-radius doesn't belong here, because similar to `width`, this class should have no opinion on styling.

File: packages/itwinui-react/src/core/Tile/Tile.tsx [hunk 17]
```diff
@@ -181,7 +181,6 @@ export const Tile = (props: TileProps) => {
         className,
       )}
       aria-disabled={isDisabled}
-      tabIndex={isActionable && !isDisabled ? 0 : undefined}
```
Comment (mayank99) [inline packages/itwinui-react/src/core/Tile/Tile.tsx:184]: This will break keyboard navigation for existing user code because (in your current version) it requires them to add `<LinkOverlay as='button'>`.

Should we automatically do that when `isActionable` is true? we can also handle disabled behavior.

File: packages/itwinui-react/src/core/utils/components/LinkBox.tsx [hunk 18]
```diff
@@ -0,0 +1,44 @@
...
+  'a',
+  // eslint-disable-next-line @typescript-eslint/ban-types
+  {}
+>;
+
+/**
+ * Polymorphic action component.
+ * It is rendered as `a` by default.
+ */
+export const LinkOverlay = React.forwardRef((props, ref) => {
```
Comment (mayank99) [inline packages/itwinui-react/src/core/utils/components/LinkBox.tsx:22]: will need tests for this.

also nit: i think filename should be `LinkOverlay` because it is the "meaty" part of the component. `LinkBox` is a simple wrapper.

File: apps/storybook/src/Tile.stories.tsx [hunk 19]
```diff
@@ -197,6 +204,46 @@ Actionable.args = {
...
+      <Tile
+        name={
+          <LinkOverlay href='/?path=/docs/core-tile--anchor-link'>
+            {name}
+          </LinkOverlay>
+        }
+        description={description}
+        metadata={metadata}
+        badge={badge}
+        thumbnail={thumbnail}
```
Comment (mayank99) [inline apps/storybook/src/Tile.stories.tsx:231]: when using arrow key navigation in screen readers, the contents of the thumbnail area (badges, quick options, img alt) will be announced first.

in https://inclusive-components.design/cards/#alternativetext, the suggestion is to use flex order to tackle this problem. can we try that?

File: packages/itwinui-css/src/utils/component-action/link-box.scss (hunk 15 above)
Comment (gretanausedaite) [reply packages/itwinui-css/src/utils/component-action/link-box.scss]: Added suggested css rules.
I think it's ok, users can see the difference in moving focus. Also screen reader does a good job showing which one is selected.

File: packages/itwinui-css/src/utils/component-action/link-box.scss (hunk 16 above)
Comment (gretanausedaite) [reply packages/itwinui-css/src/utils/component-action/link-box.scss]: Removed border radius and width.

File: packages/itwinui-react/src/core/Tile/Tile.tsx [hunk 20]
```diff
@@ -120,6 +131,21 @@ export type TileProps = {
...
+type TileActionOwnProps = {}; // eslint-disable-line
+
+export const TileAction = (
+  props: PolymorphicComponentProps<'a', TileActionOwnProps>,
+) => {
+  const tileContext = useSafeContext(TileContext);
+
+  React.useEffect(() => {
+    console.log('action useefect');
+    !tileContext.isDisabled ? tileContext.setActionable(true) : null;
```
Comment (mayank99) [inline packages/itwinui-react/src/core/Tile/Tile.tsx:143]: i've realized we can do this with `:has` and avoid an extra rerender (in new chrome/safari) and then use this approach only if `:has` is not supported. see comment here: https://github.com/iTwin/iTwinUI/pull/1127#discussion_r1134494887

also, i think there is no need to check `isDisabled` here. Tile can be both actionable and disabled at the same time, and disabled styling should take preference.

File: packages/itwinui-react/src/core/Tile/Tile.tsx [hunk 21]
```diff
@@ -137,154 +163,180 @@ export type TileProps = {
...
-        isNew={isNew}
-        status={status}
-      />
-
-      <span className='iui-tile-name-label'>{name}</span>
-    </div>
-  );
+    React.useEffect(() => {
+      console.log('tile useeffect');
+      setLocalActionable(isActionable);
```
Comment (mayank99) [inline packages/itwinui-react/src/core/Tile/Tile.tsx:200]: this use effect is not needed.

you can derive a value like this:
```js
const isActionable = isActionableProp ?? localActionable;
```

see https://beta.reactjs.org/learn/you-might-not-need-an-effect#updating-state-based-on-props-or-state

File: packages/itwinui-react/src/core/Tile/Tile.tsx [hunk 22]
```diff
@@ -137,154 +163,180 @@ export type TileProps = {
...
-      {buttons && <div className='iui-tile-buttons'>{buttons}</div>}
-    </div>
-  );
-};
+          {buttons && <div className='iui-tile-buttons'>{buttons}</div>}
+        </div>
+      </TileContext.Provider>
+    );
+  },
+  { Action: TileAction },
```
Comment (mayank99) [inline packages/itwinui-react/src/core/Tile/Tile.tsx:338]: needs jsdoc

File: packages/itwinui-react/src/core/utils/components/LinkOverlay.tsx [hunk 23]
```diff
@@ -0,0 +1,51 @@
...
+      className={cx('iui-link-overlay', className)}
+      {...rest}
+    />
+  );
+  // eslint-disable-next-line @typescript-eslint/ban-types
+}) as PolymorphicForwardRefComponent<'a', LinkOverlayOwnProps>;
+
+LinkOverlay.displayName = 'LinkOverlay';
+
+export const LinkBox = React.forwardRef((props, ref) => {
```
Comment (mayank99) [inline packages/itwinui-react/src/core/utils/components/LinkOverlay.tsx:43]: needs jsdoc

File: packages/itwinui-react/src/core/utils/components/LinkOverlay.tsx [hunk 24]
```diff
@@ -0,0 +1,51 @@
...
+
+type LinkOverlayOwnProps = {}; // eslint-disable-line @typescript-eslint/ban-types
+type LinkBoxOwnProps = {}; // eslint-disable-line @typescript-eslint/ban-types
+
+export type PolymorphicLinkOverlayProps = PolymorphicComponentProps<
+  'a',
+  LinkOverlayOwnProps
+>;
+
+export type PolymorphicLinkBoxProps = PolymorphicComponentProps<
```
Comment (mayank99) [inline packages/itwinui-react/src/core/utils/components/LinkOverlay.tsx:20]: these names should not have `Polymorphic` in them

File: packages/itwinui-react/src/core/Tile/Tile.tsx (hunk 21 above)
Comment (gretanausedaite) [reply packages/itwinui-react/src/core/Tile/Tile.tsx]: Removed useEffect

File: packages/itwinui-react/src/core/Tile/Tile.tsx (hunk 20 above)
Comment (gretanausedaite) [reply packages/itwinui-react/src/core/Tile/Tile.tsx]: Added check for ":has" functionality.

File: packages/itwinui-css/src/utils/component-action/link-box.scss (hunk 14 above)
Comment (gretanausedaite) [reply packages/itwinui-css/src/utils/component-action/link-box.scss]: I'm not wrapping Tile in LinkBox. 

Tested with MenuItem, should work:
```js
<LinkBox as={MenuItem} key='2'>
  <LinkOverlay as='button' onClick={}>Item #2</LinkOverlay>
</LinkBox>
```

File: packages/itwinui-react/src/core/Tile/Tile.tsx (hunk 17 above)
Comment (gretanausedaite) [reply packages/itwinui-react/src/core/Tile/Tile.tsx:197]: I think it should be good now. Adding isActionable class / prop.

File: packages/itwinui-react/src/core/Tile/Tile.tsx [hunk 25]
```diff
@@ -22,11 +22,35 @@ import { ProgressRadial } from '../ProgressIndicators';
...
+ * @example
+ * <Tile
+ *   name={<Tile.Action href='/new-page'>Tile name<Tile.Action/>}
+ * />
+ */
+export const TileAction = (
+  props: PolymorphicComponentProps<'a', TileActionOwnProps>,
+) => {
+  const tileContext = useSafeContext(TileContext);
+  const supportsHas = CSS.supports('selector(:has(+ *))');
```
[the remark this answers is not in the document]
Comment (gretanausedaite) [reply packages/itwinui-react/src/core/Tile/Tile.tsx]: Doing this

File: packages/itwinui-react/src/core/utils/components/PolymorphicAction.tsx (hunk 7 above)
Comment (gretanausedaite) [reply packages/itwinui-react/src/core/utils/components/PolymorphicAction.tsx]: Using own props objects.

File: packages/itwinui-react/src/core/utils/components/PolymorphicAction.tsx (hunk 6 above)
Comment (gretanausedaite) [reply packages/itwinui-react/src/core/utils/components/PolymorphicAction.tsx]: I think everything from this comment is solved.

Comment (mayank99) [reply packages/itwinui-react/src/core/utils/components/PolymorphicAction.tsx]: this comment is about naming. Have you tried to research into what other libraries are calling it and/or come up with other names yourself?

File: packages/itwinui-react/src/core/utils/components/PolymorphicAction.tsx (hunk 7 above)
Comment (mayank99) [reply packages/itwinui-react/src/core/utils/components/PolymorphicAction.tsx]: you can remove these eslint comments, they are only needed in one place.

Comment (gretanausedaite) [reply packages/itwinui-react/src/core/utils/components/PolymorphicAction.tsx]: Oops

File: packages/itwinui-react/src/core/utils/components/PolymorphicAction.tsx (hunk 6 above)
Comment (gretanausedaite) [reply packages/itwinui-react/src/core/utils/components/PolymorphicAction.tsx]: LinkBox for wrapper and LinkAction for action?
I have almost no ideas on naming. Did not find anywhere else besides chakra-ui.

Comment (mayank99) [reply packages/itwinui-react/src/core/utils/components/PolymorphicAction.tsx]: > LinkBox for wrapper and LinkAction for action?
i like it. sounds more intuitive than "overlay".

File: packages/itwinui-react/src/core/Tile/Tile.tsx [hunk 26]
```diff
@@ -137,154 +173,185 @@ export type TileProps = {
...
+              'iui-new': isNew,
+              'iui-selected': isSelected,
+              'iui-actionable': isActionable,
+              [`iui-${status}`]: !!status,
+              'iui-loading': isLoading,
+            },
+            className,
+          )}
+          aria-disabled={isDisabled}
+          onClick={onClick}
```
Comment (mayank99) [inline packages/itwinui-react/src/core/Tile/Tile.tsx:246]: don't need this onClick anymore

Comment (gretanausedaite) [reply packages/itwinui-react/src/core/Tile/Tile.tsx:246]: This onClick is "div" event and I don't want to make LinkOverlay as div.

Comment (mayank99) [reply packages/itwinui-react/src/core/Tile/Tile.tsx:246]: i dont think there is much difference in a div event vs button event.

you might able to change the type of `onClick` to something like `React.MouseEventHandler<HTMLElement>`

File: packages/itwinui-css/backstop/results/bitmaps_reference/iTwinUI_tile_Type_Action_States_dark_0_demo-action-states_0_desktop.png
Comment (mayank99) [inline packages/itwinui-css/backstop/results/bitmaps_reference/iTwinUI_tile_Type_Action_States_dark_0_demo-action-states_0_desktop.png:1]: Some of these images shouldn't have changed. Can you try reverting and see if tests pass?

File: packages/itwinui-react/src/core/Tile/Tile.tsx [hunk 27]
```diff
@@ -137,154 +174,186 @@ export type TileProps = {
...
+              'iui-folder': variant === 'folder',
+              'iui-new': isNew,
+              'iui-selected': isSelected,
+              'iui-actionable': isActionable,
+              [`iui-${status}`]: !!status,
+              'iui-loading': isLoading,
+            },
+            className,
+          )}
+          aria-disabled={isDisabled}
```
Comment (mayank99) [inline packages/itwinui-react/src/core/Tile/Tile.tsx:248]: `disabled` (or `aria-disabled` if tackling #391 too) should go on the button, not the tile

File: packages/itwinui-react/src/core/Tile/Tile.tsx [hunk 28]
```diff
@@ -10,12 +10,48 @@ import {
...
+  PolymorphicComponentProps,
+  useSafeContext,
+  getWindow,
 } from '../utils';
 import '@itwin/itwinui-css/css/tile.css';
 import { DropdownMenu } from '../DropdownMenu';
 import { IconButton } from '../Buttons';
 import { ProgressRadial } from '../ProgressIndicators';
 
+export const TileContext = React.createContext<
```
Comment (mayank99) [inline packages/itwinui-react/src/core/Tile/Tile.tsx:23]: doesn't need to be eported

File: packages/itwinui-css/src/utils/component-action/link-overlay.scss
Comment (mayank99) [inline packages/itwinui-css/src/utils/component-action/link-overlay.scss:1]: this file and the folder should be renamed to `link-action`

File: .changeset/lazy-ducks-lick.md [hunk 29]
```diff
@@ -0,0 +1,21 @@
...
+```
+
+Added `Tile.Action` to take advantage of these new a11y components in Tile.
+
+Usage:
+```js
+<Tile 
+  name={<LinkAction as='button' onClick={() => {/* Do things */}}>Tile name that is also a button</LinkAction>}
+/>
+```
```
Comment (mayank99) [inline .changeset/lazy-ducks-lick.md:21]: I think this Tile.Action part should go in a separate changeset, and the example should show Tile.Action instead of LinkAction

File: packages/itwinui-css/src/tile/tile.scss [hunk 30]
```diff
@@ -169,6 +93,97 @@ $thumbnail-selector: ':is(.iui-thumbnail-icon, .iui-tile-thumbnail-picture)';
...
+@mixin iui-selected {
+  @include iui-focus($offset: 0, $thickness: var(--iui-size-2xs));
+  --_iui-tile-status-icon-fill: var(--iui-color-icon-accent);
+
+  outline: solid 2px var(--iui-color-border-accent);
+  background-color: var(--iui-color-background-accent-muted);
+}
+
+@mixin iui-actionable {
+  @include iui-focus($offset: var(--iui-size-3xs), $thickness: var(--iui-size-3xs));
```
Comment (mayank99) [inline packages/itwinui-css/src/tile/tile.scss:105]: this `iui-focus` is no longer necessary as focus is not on the tile

File: packages/itwinui-css/src/tile/tile.scss [hunk 31]
```diff
@@ -169,6 +93,97 @@ $thumbnail-selector: ':is(.iui-thumbnail-icon, .iui-tile-thumbnail-picture)';
...
+    outline: var(--iui-size-3xs) solid var(--iui-color-border-accent);
+
+    &.iui-selected {
+      outline-width: var(--iui-size-2xs);
+    }
+  }
+
+  @supports not selector(:has(+ *)) {
+    &:where(:focus-within) {
+      outline: var(--iui-size-3xs) solid var(--iui-color-border-accent);
```
Comment (mayank99) [inline packages/itwinui-css/src/tile/tile.scss:117]: add the `iui-selected` portion in this fallback too. Right now in Firefox, there is no way to tell that a selected tile is focused

File: packages/itwinui-react/src/core/utils/components/LinkAction.tsx [hunk 32]
```diff
@@ -0,0 +1,59 @@
...
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import React from 'react';
+import cx from 'classnames';
+import {
+  PolymorphicComponentProps,
+  PolymorphicForwardRefComponent,
+} from '../props';
+
```
Comment (mayank99) [inline packages/itwinui-react/src/core/utils/components/LinkAction.tsx:11]: missing css import

File: packages/itwinui-react/src/core/Tile/Tile.tsx (hunk 27 above)
Comment (gretanausedaite) [reply packages/itwinui-react/src/core/Tile/Tile.tsx:248]: Tile disabled styling is applied with aria-disabled on Tile.
I can add aria-disabled to the button but I won't remove them from Tile. 
Yes, we can update css, but I think it's future PR problem, this one already has a lot.

File: packages/itwinui-react/src/core/utils/components/LinkAction.tsx (hunk 32 above)
Comment (gretanausedaite) [reply packages/itwinui-react/src/core/utils/components/LinkAction.tsx]: Added css import

File: packages/itwinui-css/src/tile/tile.scss (hunk 31 above)
Comment (gretanausedaite) [reply packages/itwinui-css/src/tile/tile.scss:115]: Added iui-selected portion

File: packages/itwinui-css/src/tile/tile.scss (hunk 30 above)
Comment (gretanausedaite) [reply packages/itwinui-css/src/tile/tile.scss]: Removed

File: packages/itwinui-css/src/utils/component-action/link-overlay.scss
Comment (gretanausedaite) [reply packages/itwinui-css/src/utils/component-action/link-overlay.scss:1]: renamed

File: packages/itwinui-react/src/core/Tile/Tile.tsx (hunk 28 above)
Comment (gretanausedaite) [reply packages/itwinui-react/src/core/Tile/Tile.tsx]: Is not exported anymore

File: packages/itwinui-react/src/core/Tile/Tile.tsx (hunk 27 above)
Comment (mayank99) [reply packages/itwinui-react/src/core/Tile/Tile.tsx:252]: right now the clicks are not prevented so yeah you could add `aria-disabled` on the button and conditionally pass `onClick` only when not disabled.

then there is also a bigger problem that `aria-disabled` does nothing on a generic div. you can see which roles it affects here: https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-disabled#associated_roles
so in the future we can use a data attribute instead.

File: packages/itwinui-css/backstop/results/bitmaps_reference/iTwinUI_tile_Type_Action_States_dark_0_demo-action-states_0_desktop.png
Comment (gretanausedaite) [reply packages/itwinui-css/backstop/results/bitmaps_reference/iTwinUI_tile_Type_Action_States_dark_0_demo-action-states_0_desktop.png:1]: Updated. Was missing outline-offset

**Review by elephantcatdog — APPROVED** [review]
Looks good to me!
